cucumber-spring/src/main/java/io/cucumber/spring/SpringFactory.java

```diff
@@ -82,8 +83,8 @@ private static void checkNoComponentAnnotations(Class<?> type) {
         }
     }
 
-    private static boolean hasCucumberContextConfiguration(Class<?> stepClass) {
-        return stepClass.getAnnotation(CucumberContextConfiguration.class) != null;
+    static boolean hasCucumberContextConfiguration(Class<?> stepClass) {
+        return AnnotatedElementUtils.isAnnotated(stepClass, CucumberContextConfiguration.class);
```

Comment:
Neat! Looks like this could be used to simplify the `hasComponentAnnotation` method too. Out of scope for this PR though.